src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/DirectoryBasedStoreProvider.cs

```diff
@@ -448,7 +320,7 @@ private static void EnsureDirectoryPermissions(string path, uint userId)
                 throw new CryptographicException(SR.Format(SR.Cryptography_OwnerNotCurrentUser, path));
             }
 
-            if ((dirStat.Mode & (int)Interop.Sys.Permissions.Mask) != (int)Interop.Sys.Permissions.S_IRWXU)
+            if ((dirStat.Mode & (int)Interop.Sys.Permissions.S_IRWXU) != (int)Interop.Sys.Permissions.S_IRWXU)
```

Comment:
> the intent here is that we only care that you have at least RWX owner perms, and we don't care if there are any perms beyond that, right?

Yep, just like what I told you on IM when you were making the change to directory permissions :smile:.

We could potentially change it to say "if I think I need to create the directory (as opposed to reuse it, chmod 0700", but I didn't like the notion of going behind someone and rewriting directory permissions.  And since we only fail on Add (or maybe also Remove) it was a bit weird, anyways.